src/main/java/com/auction/retailService/controller/ProductResource.java

```diff
@@ -0,0 +1,125 @@
+import com.auction.retailService.constant.RoleConstant;
+import com.auction.retailService.entity.Product;
+import com.auction.retailService.exception.UserNotExistToPerformOperationException;
+import com.auction.retailService.exception.UserUnAuthorizedException;
+import com.auction.retailService.service.ProductService;
+import com.auction.retailService.service.UserService;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+@Api(value = "Product Management API", description = "API for Product management")
+@RestController
+@RequestMapping("/retail/v1")
+public class ProductResource {
+	
+	@Autowired
+	ProductService productService;
+	
+	@Autowired
+	UserService userService;
+	
+	@ApiOperation(value = "Create Product")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Product is created Successfully"),
+            @ApiResponse(code = 400, message = "Input JSON is wrong"),
+            @ApiResponse(code =401 , message = "User is Not authorized for this operation"),
+            @ApiResponse(code = 412, message = "Validation failed Product Data."),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+	@PostMapping("/products")
+	public Product addProduct(@RequestHeader (value="userId" , required = true) Long userId , @Valid @RequestBody Product product) {
+		if(userService.isUserExist(userId)) {
+			if(userService.getUserRole(userId) == RoleConstant.RETAIL_ADMIN) {
+				return productService.addProduct(product);
+			}
+			throw new UserUnAuthorizedException(ErrorMessageConstant.USER_UNAUTHORIZED_EXCEPTION);
+		}
+		throw new UserNotExistToPerformOperationException(ErrorMessageConstant.USER_NOT_EXIST_FOR_OPERATION);
+	}
+
+	@ApiOperation(value = "Get list of Products")
+	@ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Product is retrieved Successfully"),
+            @ApiResponse(code = 400, message = "Input JSON is wrong"),
+            @ApiResponse(code =401 , message = "User is Not authorized for this operation"),
+            @ApiResponse(code = 412, message = "Validation failed Product Data."),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+	@GetMapping("/products")
+	public List<Product> getProducts()
+	{
+		return productService.getProducts();
+	}
+	
+	@ApiOperation(value = "Get Product By ProductId")
+	@ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Product is retrieved Successfully by productId"),
+            @ApiResponse(code = 400, message = "Input JSON is wrong"),
+            @ApiResponse(code =401 , message = "User is Not authorized for this operation"),
+            @ApiResponse(code = 412, message = "Validation failed Product Data."),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+	@GetMapping("products/id/{productId}")
```

Comment:
This is mapping is redundant for RESTful conventions, it should be `products/{productId}`